### DISCUSSION TITLE
Video on webplayer

### DISCUSSION THREAD

**Discussion by rainbowdesign** [AvaloniaUI/Avalonia#8186] [Q&A] [answered]
Hello community,
I am considering porting a project to avaloniaui and i like to know if i can do the things i like to.
Can i do this and how:
Play a video from a byte array or otherwise directly from code in a webplayer?

And less important:
Can i stream videos on webplayer?
Can i stream radio streams / other audio streams on webplayer, windows and android?

**Comment by maxkatz6** [comment, marked as answer]
There are two implementations of VLC control for Avalonia https://github.com/videolan/libvlcsharp and https://github.com/donandren/vlcsharpavalonia
If you need a "web" player, like a iframe of youtube page, you better to use one of the CEF controls - https://github.com/AvaloniaCommunity/awesome-avalonia#web-browsers or native webview controls (will require more work) - https://github.com/maxkatz6/AvaloniaWebView.

**Reply by rainbowdesign** [reply]
Hey, 
I looked a bit deeper now. Vlc seems to support no kind of web engine like wasm or any, as far i see it it really cannot be used on web.

One of the key targets is to play videos that are attached in savefiles and usually come in bytecode i am not so sure if it can be played in chromium when its not on a homepage but directly in code and how about the controls and interaction with chromium, is it reasonable to make a iframe like page to play attached videos and can i make the interface to it in avalonia to start and stop the video or would i have to make some kind of mini homepage there?

**Reply by maxkatz6** [reply]
So, you need to play video in the WASM app?
With latest nightly builds you can use normal html iframe inside of the avalonia app with NativeControlHost.
See example with youtube video: https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Web/EmbedSample.Browser.cs
